### PR TITLE
Handle errors for filter

### PR DESCRIPTION
**Pull request by starry97**
Part of #588

### CODE REVIEW

**Comment by starry97** [comment]
Not sure why the test failed on travis. They passed locally.

File: src/components/encoding-pane/range-filter-shelf.tsx [hunk 1]
```diff
@@ -135,6 +144,12 @@ export class RangeFilterShelfBase extends React.PureComponent<RangeFilterShelfPr
     if (renderDateTimePicker) {
       minBound = convertToDateTimeObject(minBound);
     }
+    const range = this.props.filter.range;
+    const maxBound = range[range.length - 1];
```
Comment (kanitw) [inline src/components/encoding-pane/range-filter-shelf.tsx]: range[1]?

File: src/components/encoding-pane/range-filter-shelf.tsx [hunk 2]
```diff
@@ -94,6 +94,10 @@ export class RangeFilterShelfBase extends React.PureComponent<RangeFilterShelfPr
     } else {
       range = input;
     }
+    if (range[0] > range[range.length - 1]) {
```
Comment (kanitw) [inline src/components/encoding-pane/range-filter-shelf.tsx]: range[1]?

File: src/components/field/index.tsx [hunk 3]
```diff
@@ -49,7 +49,9 @@ export interface FieldPropsBase {
 
   handleAction?: (action: FilterAction | ShelfAction | DatasetSchemaChangeFieldType) => void;
 
-  filterShow?: boolean;
+  filterShow?: {
```
Comment (kanitw) [inline src/components/field/index.tsx]: Maybe add a bit more comment what is this nested structure for.

**Pull request merged**